taxi/tests/test_views.py

```diff
@@ -0,0 +1,111 @@
+    @classmethod
+    def setUpTestData(cls):
+        Manufacturer.objects.create(name="BMW", country="Germany")
+        Manufacturer.objects.create(name="Volvo", country="Sweden")
+        cls.car1 = Car.objects.create(
+            model="M4",
+            manufacturer=Manufacturer.objects.get(id=1)
+        )
+        cls.car2 = Car.objects.create(
+            model="XC60",
+            manufacturer=Manufacturer.objects.get(id=2)
+        )
+        Driver.objects.create(
+            username="test_user_1",
+            password="Test1234!",
+            license_number="AAA00001"
+        )
+        Driver.objects.create(
+            username="test_user_2",
+            password="Test1234!",
+            license_number="AAA00002"
+        )
+
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="test_username",
+            password="Test1234!",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        response = self.client.get(MANUFACTURER_URL)
+        manufacturers = Manufacturer.objects.all()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+
+    def test_retrieve_drivers(self):
+        response = self.client.get(DRIVER_URL)
+        drivers = Driver.objects.exclude(username=self.user.username)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/driver_list.html")
+        self.assertEqual(
+            list(response.context["driver_list"]),
+            list(drivers)
+        )
+
+    def test_retrieve_cars(self):
+        response = self.client.get(CAR_URL)
+        cars = Car.objects.all()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/car_list.html")
+        self.assertEqual(
+            list(response.context["car_list"]),
+            list(cars)
+        )
+
+    def test_assign_car_to_driver(self):
+        driver = Driver.objects.get(id=self.user.id)
+        self.assertFalse(driver.cars.exists())
+
+        response = self.client.post(reverse(
+            "taxi:toggle-car-assign",
+            args=[self.car1.id]
+        ))
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(driver.cars.filter(id=self.car1.id).exists())
+
+    def test_remove_car_from_driver(self):
+        driver = Driver.objects.get(id=self.user.id)
+        driver.cars.add(self.car2)
+
+        self.assertTrue(driver.cars.filter(id=self.car2.id).exists())
+        response = self.client.post(reverse(
+            "taxi:toggle-car-assign",
+            args=[self.car2.id]
+        ))
+        self.assertEqual(response.status_code, 302)
+        self.assertFalse(driver.cars.filter(id=self.car2.id).exists())
```

Comment:
In the `test_assign_car_to_driver` and `test_remove_car_from_driver` methods, the existence of the car in the driver's car list is being checked before and after the request. This is a good practice as it ensures that the car was correctly assigned or removed. However, it would be better to also check the response context to make sure that the changes are correctly reflected in the response.